src/services/device.service.ts

```diff
@@ -0,0 +1,31 @@
+import { db } from '../database';
+
+export interface Device {
+  userId: number;
+  deviceToken: string;
+}
+
+interface InsertDeviceResult{
+  user_id: number;
+  device_token: string;
+}
+
+interface DeviceCreated{
+  user_id: number;
+  device_token: string;
+}
+
+const addDevice = async (device: Device): Promise<DeviceCreated> => {
+  const result = await db.sql<InsertDeviceResult>('insert-devices', {
+    user_id: device.userId,
+    device_token: device.deviceToken,
+  });
+  return {
+    user_id: result.rows[0].user_id,
+    device_token: result.rows[0].device_token,
```

Comment:
Let's try to match the input here: the client sends camelCased names (`userId`, `deviceToken`), but the database wants snake_cased names. We should handle that impedance mismatch:

```suggestion
    userId: result.rows[0].user_id,
    deviceToken: result.rows[0].device_token,
```

That'd also allow us to return a `Device` instead of a `DeviceCreated`, which means we can delete the `DeviceCreated` interface.